### PR TITLE
Added react ui filters to users, roles, tenants

### PR DESCRIPTION
**Pull request by yekalkan**

### CODE REVIEW

**Comment by ryoldash** [comment]
@hikalkan It would be an honor :)

File: reactjs/src/scenes/Roles/index.tsx [hunk 1]
```diff
@@ -24,21 +24,29 @@ const confirm = Modal.confirm;
 @inject(Stores.RoleStore)
 @observer
 class Role extends AppComponentBase<IRoleProps> {
```
Comment (ryoldash) [inline reactjs/src/scenes/Roles/index.tsx]: Can you modify IRoleState as: 
export interface IRoleState {
  modalVisible: boolean;	  
  maxResultCount: number;	  
  skipCount: number;	  
  roleId: number;
  filter: string;
}	

and modify component definition as 
class Role extends AppComponentBase<IRoleProps, IRoleState>

File: reactjs/src/scenes/Roles/index.tsx [hunk 2]
```diff
@@ -101,7 +109,12 @@ class Role extends AppComponentBase<IRoleProps> {
 
   saveFormRef = (formRef: any) => {
     this.formRef = formRef;
-  };
+    };
+
+    handleChange(event: any) {
+        this.state.filter = event.target.value;
```
Comment (ryoldash) [inline reactjs/src/scenes/Roles/index.tsx]: You can change it arrow function following:

handleChange = (event: any) => {
    this.setState({ filter: event.target.value });
}

If Search component used then 
handleSearch = (event: any) => {
    this.setState({ filter: event.target.value });
}

File: reactjs/src/scenes/Roles/index.tsx [hunk 3]
```diff
@@ -159,6 +172,15 @@ class Role extends AppComponentBase<IRoleProps> {
           >
             <Button type="primary" shape="circle" icon="plus" onClick={() => this.createOrUpdateModalOpen({ id: 0 })} />
           </Col>
+        </Row>        
+        <Row>
+          <Col sm={{ span: 5, offset: 0 }}>
```
Comment (ryoldash) [inline reactjs/src/scenes/Roles/index.tsx]: AntDesign has Search Component. You could use the component as follows: 
const Search = Input.Search;

<Search
      placeholder={L('SearchPlaceHolder')}
      onSearch={this.handleSearch}
      enterButton
    />

File: reactjs/src/scenes/Roles/index.tsx (hunk 2 above)
Comment (ryoldash) [reply reactjs/src/scenes/Roles/index.tsx]: If Search component used then can be modified as follows:
handleSearch = (value: string) => {
this.setState({ filter: value }, async () => await this.getAll());
}

**Comment by yekalkan** [comment]
Hi @ryoldash 

I've just tried to do everything you mentioned, but every one of them broke the app someway. I have zero experience on react and error messages aren't very helpfull. so please could you add a commit and fix those things?

**Comment by ryoldash** [comment]
@yekalkan I have finished code refactoring. 

Can I have write access to Ui-Filters-React branch? Or can you guide me to push my local changes to your branch?

roles
permission denied

**Comment by acjh** [comment]
You can submit a PR to Ui-Filters-React branch after pushing to your own fork.
You can use this compare link: https://github.com/aspnetboilerplate/module-zero-core-template/compare/Ui-Filters-React...ryoldash:Ui-Filters-React

**Comment by ryoldash** [comment]
@yekalkan @acjh I have created PR to Ui-Filters-React branch.

**Comment by yekalkan** [comment]
@ryoldash thanks you.

**Pull request merged**